lib/pbench/test/unit/server/query_apis/test_query_builder.py

```diff
@@ -31,72 +35,131 @@ def current_user(self) -> User:
-                    {"term": {"icecream": "chocolate"}},
-                    {"term": {"authorization.owner": id}},
-                ]
-            }
-        }
-
-    def test_user_admin(self, elasticbase, server_config, current_user_admin):
+    @pytest.mark.parametrize(
+        "ask",
+        [
+            ({}, {}),
+            ({"access": "public"}, {"access": "public"}),
+            ({"access": "private"}, {"access": "private"}),
+            ({"user": USER_ID}, {"user": USER_ID}),
+            (
+                {"user": USER_ID, "access": "private"},
+                {"user": USER_ID, "access": "private"},
+            ),
+            (
+                {"user": USER_ID, "access": "public"},
+                {"user": USER_ID, "access": "public"},
+            ),
+        ],
+    )
+    def test_admin(self, elasticbase, server_config, current_user_admin, ask):
+        term = {"term": {"icecream": "ginger"}}
+        query = elasticbase._get_user_query(ask[0], [term])
+        filter = [term]
+        if "access" in ask[1]:
+            filter.append({"term": {"authorization.access": ask[1]["access"]}})
+        if "user" in ask[1]:
+            filter.append({"term": {"authorization.owner": ask[1]["user"]}})
+        assert query == {"bool": {"filter": filter}}
+
+    @pytest.mark.parametrize(
+        "ask",
+        [
+            ({"access": "public"}, {"access": "public"}),
+            ({"access": "private"}, {"user": SELF_ID, "access": "private"}),
+            ({"user": SELF_ID}, {"user": SELF_ID}),
+            ({"user": USER_ID}, {"user": USER_ID, "access": "public"}),
+            (
+                {"user": SELF_ID, "access": "private"},
+                {"user": SELF_ID, "access": "private"},
+            ),
+            (
+                {"user": SELF_ID, "access": "public"},
+                {"user": SELF_ID, "access": "public"},
+            ),
+            (
+                {"user": USER_ID, "access": "public"},
+                {"user": USER_ID, "access": "public"},
+            ),
+        ],
+    )
+    def test_auth(self, elasticbase, server_config, current_user_drb, ask):
+        """
+        Test the query builder when we have an authorized user.
+
+        NOTES:
+
+        1) We don't test {} here: that's left to a separate test case rather
+           than building the unique disjunction syntax here;
+        2) We don't test {"user": USER_ID, "access": "private"} here:
+           That raises the AssemblyError exception, and is tested in a
+           separate test case.
+        """
+        term = {"term": {"icecream": "ginger"}}
+        query = elasticbase._get_user_query(ask[0], [term])
+        filter = [term]
+        if "access" in ask[1]:
+            filter.append({"term": {"authorization.access": ask[1]["access"]}})
+        if "user" in ask[1]:
+            filter.append({"term": {"authorization.owner": ask[1]["user"]}})
+        assert query == {"bool": {"filter": filter}}
```

Comment:
Should this be refactored into a helper function?

Reply:
Before I started, I thought about going back to refactor; when I was done I just wanted it to be done and didn't even think about it. The bit about being able to parametrize "x,y" is actually new to me, and if I do that I'll need to take a hacksaw to all of this code anyway, so I'll think about it. It's already clear this PR will *never* be done anyway since I can't even read one set of comments before another set drops!

Reply:
> I can't even read one set of comments before another set drops!

I apologize on behalf of GitHub for that.  I typically do reviews with two tabs open in my browser -- one to the "Conversations" view and one to the "Files" view.  Once you start a review, then every comment you enter after that becomes part of the review; if you refresh the "Conversations" view after starting a review in the "Files" view, then all of the "Reply" text boxes disappear (because it then knows that you have a review "in progress"), and you cannot enter any comments from that page; however, if you _don't_ refresh the "Conversations" view after starting a review, then you can still enter comments...but doing so submits the whole review, including any comments that you've entered in the other tab...and it does so without updating the other tab (which is why I made that cryptic comment earlier -- because I couldn't find half the comments that I thought I'd entered into my review!).  (I generally refresh my "Conversations" tab just before submitting my review because it displays all of my comments in one place (marked as "pending"), so that I can review them.)

I was halfway through my review, and I saw something in the "Conversations" view that I wanted to comment on directly, outside the context of my review, but hitting the "Comment" button committed my incomplete review...and so, once I submitted my review, you then saw a second batch of comments arrive.  That was not the presentation that I had intended.